API.md

```diff
@@ -1,323 +1,354 @@
+<!-- or {{ your.Scope.Name }} -->
+<input type="text" name="number" ng-model="form.number" validator="number" no-validation-message="{{ noValidationMessage }}"/>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+### **Add Valid Callback Function,** `invalid-callback` **&** `valid-callback`
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ html
+<label>Required (Invalid Callback alert)</label>
+<input type="text" name="requiredCallback" ng-model="form.requiredCallback" validator="required" invalid-callback='error("Must be Required");'/>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+### **Select the validation method** `watch blur submit submit-only`**, default as** `watch` 
+
+`validationProvider.validate(form).success(callback).error(callback)` use
+callback to continue your submit
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ html
+<label>Watch method</label>
+<input type="text" name="number" ng-model="form.number" validator="number" valid-method="watch"/>
+<input type="text" name="number" ng-model="form.number" validator="number"/>
+
+<label>Blur method</label>
+<input type="text" name="number" ng-model="form.number" validator="number" valid-method="blur"/>
+<!-- or try ng-model-options="{ updateOn: 'blur' }" -->
+<input type="text" name="number" ng-model="form.number" validator="number" ng-model-options="{ updateOn: 'blur' }"/>
+
+<label>Submit method</label>
+<form name="Form">
+    <input type="text" name="number" ng-model="form.number" validator="number" valid-method="submit"/>
+    <button ng-click="form.submit(Form)"></button>
+</form>
+
+<label>Submit Only method</label>
+<form name="Form">
+    <input type="text" name="number" ng-model="form.number" validator="number" valid-method="submit-only"/>
+    <button ng-click="form.submit(Form)"></button>
+</form>
+
+<script>
+    // ... validate method, it will check `checkValid(Form)`
+    $scope.form = {
+        submit: function (form) {
+            $validationProvider.validate(form)
+                .success(successCallback)
+                .error(errorCallback);
+        }
+    };
+    // ...
+</script>
+
+<!-- In angular validation 1.2.0
+     More easy, more clean in your js code -->
+
+<form name="Form">
+    <input type="text" name="number" ng-model="form.number" validator="number" valid-method="submit-only"/>
+    <button validation-submit="Form" ng-click="form.submit()"></button>
+</form>
+
+<script>
+    // Don't need include $validationProvider.validate() anymore
+    $scope.form = {
+        submit: function () {
+            // your ng-click success callback
+        }
+    };
+</script>
+
+<!-- Clean, right ? -->
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ 
+
+### **Select a global validation method** `watch blur submit submit-only`**
+
+`validationProvider.setValidMethod('submit')`

```

Comment:
Sorry, I should say more about this.
I need this section changes only.

We should keep the original documentation style

means we don't need ~~~~~~~~~ style.